osu.Game/Screens/Play/Player.cs

```diff
@@ -955,7 +955,7 @@ private bool onFail()
                     ScoreProcessor.FailScore(Score.ScoreInfo);
                     OnFail();
 
-                    if (GameplayState.Mods.OfType<IApplicableFailOverride>().Any(m => m.RestartOnFail && m.TriggeredFail))
+                    if (GameplayState.Mods.OfType<IApplicableFailOverride>().Any(m => m.RestartOnFail && m == HealthProcessor.FailTrigger))
```

Comment:
Why not just do `HealthProcessor.FailTrigger.RestartOnFail`

Reply:
you are right will change that after making changes